cadc-gms/src/main/java/org/opencadc/auth/PosixMapperClient.java

```diff
@@ -0,0 +1,150 @@
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.auth;
+
+import ca.nrc.cadc.auth.PosixPrincipal;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+import javax.security.auth.Subject;
+import org.apache.log4j.Logger;
+import org.opencadc.gms.GroupURI;
+
+/**
+ * Stub to figure out API.
+ * 
+ * @author pdowler
+ */
+public class PosixMapperClient {
+    private static final Logger log = Logger.getLogger(PosixMapperClient.class);
+
+    private final URI resourceID;
+    
+    public PosixMapperClient(URI resourceID) {
+        this.resourceID = resourceID;
+    }
+    
+    // use case: cavern does this when caller tried to set group permissions
+    // if we have multiple, this is convenience
+    // intent: this may create and persist a local GID as a side effect
+    public Integer getGID(GroupURI guri) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // or check multiple in one call??
+    public List<PosixGroup> getGID(List<GroupURI> groups) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: cavern does this when reading a node from disk and output the node doc
+    // if we have multiple, this is convenience
+    public GroupURI getURI(Integer gid) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // or check multiple in one call??
+    public List<PosixGroup> getURI(List<GroupURI> groups) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: cavern needs PosixPrincipal in the subject
+    // use case: StandardIdentityManager calls this to add local identity
+    public void augment(Subject s) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: skaha needs to complete username-uid map for user containers
+    public Iterator<PosixPrincipal> getUserMap() {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: skaha needs the complete groupname-gid map for user containers
+    public Iterator<PosixGroup> getGroupMap() {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: skaha getGroupMap response - probably extracted into a normal class
+    // depending on where/how we deal with user default group
+    public static class PosixGroup {
```

Comment:
If all goes according to plan, then the Group entry would also have the members:

`CHIMERWX:x:9999:jenkinsd,adamian,sgaudet`

Reply:
This is supposed to just be the mapping of name <-> gid... does it really need the list of members here? That won't come from the mapper service - it comes from a GMS service. But it is a good point... it changes the complexion of the injected db a lot. Need to discuss.

Reply:
moving discussion to `team-src-red` slcack channel